dashboard/dashboard-service.js

```diff
@@ -33,6 +37,9 @@ function createDashboardService(ownerId) {
             });
         }
         const templateData = dataset.reduce((acc, questionnaireData) => {
+            const resumeLink = questionnaireData.analyticsId
+                ? `/apply/resume/${questionnaireData.questionnaireId}?a_id=${questionnaireData.analyticsId}`
```

Comment:
instead of passing the `analyticsId` around so that the resume handler can get it, can you not just get it from the session, or from the stored value in the db?

Reply:
You couldn't rely on getting it from the session because there could already be one in the session when you try to resume an application. Plus the dashboard can have multiple applications, you can only set it in the session when you click the link to resume, not when it's initially retrieved. 

Also, I felt passing it here was the better option as it reduced the number of calls made to the db. The `metadata` endpoint is called when you land on the dashboard. If a user resumes an application from the dashboard, you could either pass the ID as I have done, or call the metadata endpoint again on the `/resume` route.